src/include/simeng/Core.hh

```diff
@@ -13,6 +14,40 @@ class ArchitecturalRegisterFileSet;
 
 enum CoreStatus : uint8_t { halted, idle, executing, switching };
 
+/** The result from a handled exception. */
+struct ExceptionResult {
+  /** Whether execution should halt. */
```

Comment:
A bit more information on why the execution will halt would be better. It's not clear as the variable name is `fatal`

Reply:
Reworded

Reply:
Should it say : Whether the outcome of the **execution**  or **exception** ........

Reply:
Good spot, should be the latter